spec/controllers/nodes_controller_spec.rb

```diff
@@ -95,5 +95,19 @@
       end
       expect(response.status).to eq 302
     end
+
+    context "a role fails to be assigned" do
+      # rubocop:disable AnyInstance
+      before do
+        allow_any_instance_of(Minion).to receive(:assign_role).and_return(false)
```

Comment:
I couldn't find a better way to mock this response in a stylish way without messing too much with the inners of the bootstrap method. Maybe this is a code smell, but if you come up with a better mocking I'd be happy to hear about it.